rules/flags/index.bzl

```diff
@@ -1,7 +1,8 @@
 load("@bazel_skylib//rules:common_settings.bzl", "BuildSettingInfo")
 
 def _write_flag_to_file_impl(ctx):
-    out = ctx.actions.declare_file(ctx.attr.name)
+    out_file = ctx.attr.out if ctx.attr.out else ctx.attr.name
+    out = ctx.actions.declare_file(out_file)
```

Comment:
wdyt about making `out` a required arg rather than defaulting to `name`, so that we don't hit this again in the future?

Reply:
I would need to refactor other places that use this rule to follow that convention.

Let me give it a try to see if it's worth it. I am also willing to change Gazelle code if the convention does not fit our use case :P.